File: backend/src/Workora.Application/Features/Notifications/DTOs/NotificationDto.cs
using Workora.Domain.Enums;
using Workora.Shared.Responses;

namespace Workora.Application.Features.Notifications.DTOs;

/// <summary>
/// DTO representing an in-app user notification.
/// </summary>
public record NotificationDto(
    int Id,
    Guid Uuid,
    int UserId,
    string Title,
    string Message,
    NotificationType Type,
    string? ActionUrl,
    bool IsRead,
    DateTimeOffset? ReadAt,
    DateTimeOffset CreatedAt);

File: frontend/workora-ui/src/app/data/dtos/notification.dto.ts
export interface NotificationDto {
  id: number;
  uuid: string;
  userId: number;
  title: string;
  message: string;
  type: string;
  actionUrl?: string | null;
  isRead: boolean;
  readAt?: string | null;
  createdAt: string;
}

export interface UnreadNotificationCountDto {
  unreadCount: number;
}
